src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -164,7 +162,7 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
                 default => $this->resolveCustomIdentifier($node->name, $typeContext),
             };
 
-            if ($type instanceof ObjectType && \in_array($type->getClassName(), self::COLLECTION_CLASS_NAMES, true)) {
+            if ($type instanceof ObjectType && (is_a($type->getClassName(), \Traversable::class, true) || is_a($type->getClassName(), \ArrayAccess::class, true))) {
```

Comment:
is_subclass_of?

Reply:
When using `is_subclass_of`, only subclasses will return true. And I think that it'd be great to be able to understand `Traversable` as a collection as well. WDYT?

Reply:
Not sure what you mean. is_subclass_of works with interfaces too: https://3v4l.org/bP6JN

Reply:
Exactly, thanks @xabbuh.